src/prefect/utilities/configuration.py

```diff
@@ -27,7 +27,8 @@ def set_temporary_config(temp_config: dict) -> Iterator:
         for key, value in temp_config.items():
             prefect.config.set_nested(key, value)
 
-        yield prefect.config
+        with prefect.context(config=prefect.config):
```

Comment:
Just for clarity, do you mind throwing a comment over this
```suggestion
        # ensure the new config is available in context
        with prefect.context(config=prefect.config):
```